frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/helpers/TestClientHelper.java

```diff
@@ -27,15 +29,24 @@ public static boolean isTestClientWithAllowedEmail(
                         .orElseThrow(() -> new ClientNotFoundException(userContext.getSession()));
 
         var isTestClientWithAllowedEmail =
-                clientRegistry.isTestClient()
-                        && clientRegistry
-                                .getTestClientEmailAllowlist()
-                                .contains(userContext.getSession().getEmailAddress());
+                (clientRegistry.isTestClient()
+                        && emailMatchesAllowlist(
+                                userContext.getSession().getEmailAddress(),
+                                clientRegistry.getTestClientEmailAllowlist()));
 
         LOG.info(
                 "Is request from a test client with a test client email address: {}",
                 isTestClientWithAllowedEmail);
 
         return isTestClientWithAllowedEmail;
     }
+
+    public static boolean emailMatchesAllowlist(String emailAddress, List<String> regexAllowList) {
+        for (String regex : regexAllowList) {
+            if (Pattern.compile(regex).matcher(emailAddress).find()) {
```

Comment:
```suggestion
            if (Pattern.matches(regex, emailAddress)) {
```

Reply:
It also might be worth putting a try {} catch {} here in case the regex is malformed else this will crash out.